src/main/java/frc/robot/commands/vision/TestVision.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.commands.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.Vision;
+
+public class TestVision extends CommandBase{
+  private Drivetrain m_drive;
+  private Vision m_vision;
+  private Translation2d m_visionStartTranslation, m_driveStartTranslation;
+  private Pose2d m_currentPose = null;
+  private int m_endCounter = 0;
+  private int m_printCounter=0;
```

Comment:
I think it might be a good idea to use a timer rather than counting cycles. It would proobly be easier to understand and easier to adjust. Instead of printing out your result every few cycles it might be a good idea to use shuffleboard.